telegram-history-dump.rb

```diff
@@ -129,9 +135,13 @@ def dump_dialog(dialog)
       end
 
       if dump_msg
-        process_media(dialog, msg)
-        if $dumper.dump_msg(dialog, msg) == false
-          keep_dumping = false
+        begin
+          process_media(dialog, msg)
+          if $dumper.dump_msg(dialog, msg) == false
+            keep_dumping = false
+          end
+        rescue Timeout::Error
+          keep_dumping = true
```

Comment:
I just gave it a deeper look, it seems the `rescue StandardError => e` is not rescuing the `Timeout::Error` because it's inside the `timeout` call (and runs on a separate thread, while the error is thrown on the main thread).

So a cleaner alternative to @pdonadeo's solution would be to just put the `Timeout::timeout` call inside of the `begin .. rescue` block.